lib/chef/provider/directory.rb

```diff
@@ -64,7 +64,13 @@ def define_resource_requirements
               is_parent_writable = lambda do |base_dir|
                 base_dir = ::File.dirname(base_dir)
                 if ::File.exists?(base_dir)
-                  Chef::FileAccessControl.writable?(base_dir)
+                  if Chef::FileAccessControl.writable?(base_dir)
+                    true
+                  elsif Gem::Version.new(node['platform_version']) >= Gem::Version.new('10.11') && Chef::Util::PathHelper.is_sip_path?(base_dir)
```

Comment:
You can always pass the node object into `is_sip_path?` - depending on how many call sites there are to update...